### PR TITLE
make piping work and other CLI improvements

### PR DESCRIPTION
**Pull request by songgao**
(this includes the `keybase chat send` behaviors that we just talked about on slack)

```
songgao at Song-ThinkPad in ~GOPATH/src/github.com/keybase/client/go/keybase (songgao/CORE-3809)
$ ./keybase chat send
▶ WARNING Running in devel mode
There are 7 conversations. Please choose one:
[1] t_alice,test2                                                                                                                                                                              
[2] t_charlie,test2                                                                                                                                                                            
[3] t_bob,test2                                                                                                                                                                                
[4] t_alice,t_bob,test2                                                                                                                                                                        
[5] t_bob,t_charlie,test2                                                                                                                                                                      
[6] t_bob,t_charlie,t_doug,test2                                                                                                                                                               
[7] t_doug,test2                                                                                                                                                                               
Please enter a number [1-7]: 1
Send to [t_alice,test2]? Hit Ctrl-C to cancel, or enter message content to send: hello!

songgao at Song-ThinkPad in ~GOPATH/src/github.com/keybase/client/go/keybase (songgao/CORE-3809)
$ ./keybase chat send t_charlie
▶ WARNING Running in devel mode
Please enter message content: hello?

songgao at Song-ThinkPad in ~GOPATH/src/github.com/keybase/client/go/keybase (songgao/CORE-3809)
$ echo hello stdin | ./keybase chat send --stdin
Error parsing command line arguments: --stdin requires 1 argument [conversation]

songgao at Song-ThinkPad in ~GOPATH/src/github.com/keybase/client/go/keybase (songgao/CORE-3809)
$ echo hello stdin | ./keybase chat send --stdin t_bob
▶ WARNING Running in devel mode

```

### CODE REVIEW

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits 1df2ac861be0...3db785cbb99c introduced some visual changes on linux:
 * Changed: **New_Proof_Post-Facebook** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1474499225441-1df2ac861be0-3db785cbb99c-linux/New_Proof_Post-Facebook-1df2ac861be0.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1474499225441-1df2ac861be0-3db785cbb99c-linux/New_Proof_Post-Facebook-3db785cbb99c.png)  
   New_Proof_Post-Facebook comparison

File: go/client/cmd_chat_send.go [hunk 1]
```diff
@@ -74,8 +81,20 @@ func (c *cmdChatSend) Run() (err error) {
...
-	if err = c.G().UI.GetTerminalUI().PromptForConfirmation(fmt.Sprintf("Send to %s?", conversationInfo.TlfName)); err != nil {
-		return err
+	switch {
+	case userChosen && len(c.message) == 0:
+		c.message, err = c.G().UI.GetTerminalUI().Prompt(PromptDescriptorEnterChatMessage, fmt.Sprintf("Send to [%s]? Hit Ctrl-C to cancel, or enter message content to send: ", conversationInfo.TlfName))
+		if err != nil {
+			return err
+		}
+	case userChosen:
+		return errors.New("\"this should never happen\"")
```
Comment (mmaxim) [inline go/client/cmd_chat_send.go]: Maybe a better message here?

Comment (songgao) [inline go/client/cmd_chat_send.go]: That was supposed to be a joke; but I'll switch it out to be more descriptive :D

**Pull request merged**